app/controllers/authenticate_controller.rb

```diff
@@ -44,27 +64,34 @@ def authn_jwt_status
   end
 
   def update_config
-    body_params = Rack::Utils.parse_nested_query(request.body.read)
-
     Authentication::UpdateAuthenticatorConfig.new.(
-      account: params[:account],
-      authenticator_name: params[:authenticator],
-      service_id: params[:service_id],
-      username: ::Role.username_from_roleid(current_user.role_id),
-      enabled: body_params['enabled'] || false
+      account: update_config_input.account,
+      authenticator_name: update_config_input.authenticator_name,
+      service_id: update_config_input.service_id,
+      username: update_config_input.username,
+      enabled: update_config_input.enabled
```

Comment:
a better fix here will be to pass `update_config_input` to `UpdateAuthenticatorConfig` and extract the fields there, like we do in `ValidateStatus` and `Authentication::Authenticate`